Improve trader mission - Only give players vague location until found

jwoodruff40 opened the pull request:
## What type of PR is this?
1. [ ] Bug
2. [x] Change
3. [ ] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:

This PR makes locating the trader more difficult. When trader mission is spawned, only a vague general AO where the trader operates is shown on the map. Players will have to search this area to actually *find* the trader. Kind of reinforces his discrete nature, and should make it more difficult / dangerous to actually find and get to him. The task icon is set to the center of this AO.

Once located, the standard marker with the exact location of the trader is added to the map and the vague one removed.

![traderMarkerVague](https://github.com/user-attachments/assets/a034e10a-d8a1-4327-b86e-de650e9d32f9)


### Please specify which Issue this PR Resolves (If Applicable).
None.

### Please verify the following.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

********************************************************
Notes:

The vague marker size has a radius of ~~1500m~~ 750m. ~~Maybe *slightly* big, but works well on Altis given how open it generally is. Probably would be a massive pain searching an area of that size on a dense / highly-vegetated map like CLN or Tanoa though. Not sure what the best solution to that problem is.~~

jwoodruff40 committed:
Only show general location of trader for locate trader task

jwoodruff40 committed:
Increase traderMarkerVague size. Don't show exact position marker until found.

Silence-112 reviewed (commented):
Looks alright, have you tested the vague marker after a restart?

Also on the topic of it being a pain to find, perhaps the trader could have a permanent smoke grenade or something similar at the door

jwoodruff40 commented:
> have you tested the vague marker after a restart?

Actually no, will do.


> Also on the topic of it being a pain to find, perhaps the trader could have a permanent smoke grenade or something similar at the door

Will need to look into how to do this, shouldn't be *too* difficult. Can probably steal the code from the collect airdrop mission or something. I presume the idea is the smoke would go away once found since you now have an exact map marker for the trader location?

Silence-112 commented:
> I presume the idea is the smoke would go away once found since you now have an exact map marker for the trader location?

Precisely

jwoodruff40 committed:
Fix trader marker persistence; reduce vague marker size; add smoke to trader location

jwoodruff40 commented:
- Fixed trader maker persistence (after finding the trader)
 - Reduced radius of vague marker to 750m
 - Added smoke (burning barrel) to trader location to make it somewhat easier to find when in the general AO

Far away:
![trader smoke far](https://github.com/user-attachments/assets/6c5070bd-284c-48bd-8b8b-f5712681545b)

Near:
![trader smoke near](https://github.com/user-attachments/assets/28750291-821e-4320-b80a-5335658f7cef)

The intensity of the smoke may be a little much, but wood smoke is too light to be of much use in most environmental conditions even though it looks better. I guess the lore here is the trader is burning oil to make himself visible knowing that you're coming lol. Smoke grenade smoke is also too ground-hugging to be of much use. Obviously in any case, effect is even more exaggerated if using mods like blastcore, but we can't support everything...

jwoodruff40 committed:
Merge branch 'unstable' into improve-trader-mission

jwoodruff40 committed:
fack

Silence-112 reviewed (approved):
Worked fine in testing from what I can tell, looks fine
